### DISCUSSION TITLE
RVF v6 with Zod and Zod Form Data?

### DISCUSSION THREAD

**Discussion by RobertGemmaJr** [airjp73/rvf#399] [Q&A]
My group is currently in the process of migrating from `remix-validated-form` to `rvf`. We use the `withZod` validation and the `zfd` preprocessors pretty extensively.

My question is simple - is the `zfd` library still necessary? The pre-processing of text/file/number/etc. done for required fields is really nice but I'm not sure if they'll be necessary after the migration.

The v6 documentation doesn't mention it at all so I assume it's no longer necessary? Does anyone else have experience with this?

**Comment by airjp73** [comment]
It's definitely still relevant, but `zod` does a better job at handling some of the most common cases out of the box, now. I think it's personal preference though. I've never really loved the API where you have to pass the schema into the helpers because there are gatchas. `zfd.text().optional()` looks valid, for example, but `zfd.text(z.string().optional())` is correct.

In my own code I've pretty much stopped using `zfd.numeric` and `zfd.checkbox` in favor of `z.coerce.number` and `z.coerce.boolean`. I've also started liking `z.string().min(1, 'message')` more than`zfd.text(z.string({ required_error: 'message' }))`.

However, I often pull in `zfd` in order to use `zfd.formData` in places where I have FormData, URLSearchParams, etc but it isn't a zfd form. And zod doesn't really have a built-in answer for `repeatable`, `json`, etc either.

It's probably still a good idea to port the docs over to the new site. I was considering pulling it into a separate repo, which is why I didn't do that yet, I think.
